doc/c-style-guide.md

```diff
@@ -0,0 +1,41 @@
+GRPC C STYLE GUIDE
+=====================
+
+Background
+----------
+
+Here we document style rules for C usage in the gRPC Core library.
+
+General
+-------
+
+- Layout rules are defined by clang-format, and all code should be passed through
+  clang-format. A (docker-based) script to do so is included in 
+  tools/distrib/clang_format_code.sh.
+
+Header Files
+------------
+
+- Public header files (those in the include/grpc tree) should compile as pedantic C89
```

Comment:
I personally think the likelihood of someone having a enforced-C89-only code base who would also want to use gRPC is vanishingly small. Is it really worth the extra bother of having to ensure headers are C89, *and* losing the C99 features (I want bool!) to optimize for that?